app/src/main/java/org/glucosio/android/db/DatabaseHandler.java

```diff
@@ -90,6 +90,95 @@ public void updateUser(User user) {
         realm.commitTransaction();
     }
 
+    public boolean addReminder(Reminder reminder) {
+        // Check for duplicates first
+        if (getReminder(reminder.getId()) == null) {
+            realm.beginTransaction();
+            realm.copyToRealm(reminder);
+            realm.commitTransaction();
+            return true;
+        }
+
+        return false;
+    }
+
+    public void updateReminder(Reminder reminder) {
+        realm.beginTransaction();
+        realm.copyToRealmOrUpdate(reminder);
+        realm.commitTransaction();
+    }
+
+    public void deleteReminder(Reminder reminder) {
+        realm.beginTransaction();
+        reminder.deleteFromRealm();
+        realm.commitTransaction();
+    }
+
+    public void deleteReminder(long id) {
+        deleteReminder(getReminder(id));
+    }
+
+    public boolean areRemindersActive() {
+        ArrayList<Date> activeAlarmsDates = new ArrayList<>();
+        RealmResults<Reminder> activeRemindersList =
+                realm.where(Reminder.class)
+                        .equalTo("active", true)
+                        .findAll();
+
+        for (int i=0; i<activeRemindersList.size(); i++){
+            activeAlarmsDates.add(activeRemindersList.get(i).getAlarmTime());
+        }
+
+        return activeAlarmsDates.size() > 0;
```

Comment:
Why do you need to add dates to the list? Why do not to use reminders list size? Does Realm have count function?